src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -300,12 +323,148 @@ def finalize(self) -> RankBasedMetricResults:  # noqa: D102
             metrics=self.metrics,
             rank_and_candidates=_iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates),
         )
+        if not self.clear_on_finalize:
+            return result
+
         # Clear buffers
         self.ranks.clear()
         self.num_candidates.clear()
 
         return result
 
+    def finalize_multi(self, num: int = 1_000, seed: int = 42) -> Mapping[str, Sequence[float]]:
+        """Bootstrap from :meth:`finalize`.
+
+        :param num:
+            the number of resampling steps
+        :param seed:
+            the random seed.
+
+        :return:
+            a flat dictionary from metric names to list of values
+        """
+        result: DefaultDict[str, List[float]] = defaultdict(list)
+
+        for i in range(num):
+            generator = numpy.random.default_rng(seed=seed + i)
+
+            def resample(
+                entry: Tuple[ExtendedTarget, RankType, np.ndarray, np.ndarray, Optional[np.ndarray]],
+                generator=generator,
+            ) -> Tuple[ExtendedTarget, RankType, np.ndarray, np.ndarray, Optional[np.ndarray]]:
+                """Resample ranks.
+
+                :param entry:
+                    the rank-pack
+                :param generator:
+                    the random number generator instance
+
+                :return:
+                    a re-sampled pack
+                """
+                target, rank_type, ranks, candidates, weights = entry
+                n = len(ranks)
+                ids = generator.integers(n, size=(n,))
+                return target, rank_type, ranks[ids], candidates[ids], None if weights is None else weights[ids]
+
+            single_result = RankBasedMetricResults.from_ranks(
+                metrics=self.metrics,
+                rank_and_candidates=map(resample, _iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates)),
```

Comment:
there's `itt.starmap` as well if you want the function to be applied like [f(*t) for t in iter_ranks(...)]

Reply:
https://github.com/pykeen/pykeen/pull/1084/commits/dbea25ceda269d9f2031333769bf3eb97e3ee43c